### PR TITLE
[Refactor] More unravel fixes

### PR DESCRIPTION
**Pull request by vmoens**
The problem this PR solves is that we made changes in the way unravel_key(s) work but we'll revert that in tenrordict.
In the meantime, torchrl has been adapted.
Rather than reverting the PR in torchrl which will break the CI, we will do the following:
  - Merge this PR in torchrl to emulate the behaviour of unravel_key in tensordict after https://github.com/pytorch-labs/tensordict/pull/473 is merged
  - Merge https://github.com/pytorch-labs/tensordict/pull/473 in tensordict
  - Revert imports in torchrl to tensordict instead of emulated functions

I expect this will temporarily impact negatively the benchmarks but they should recover afterwards.

cc @matteobettini

### CODE REVIEW

File: torchrl/_utils.py [hunk 1]
```diff
@@ -529,3 +531,39 @@ def clone(self):
...
+    for key in key_list:
+        key = unravel_keys(key)
+        if isinstance(key, tuple) and len(key) == 1:
+            key_list_out.append(key[0])
+        else:
+            key_list_out.append(key)
+    return key_list_out
+
+
+def unravel_keys(key):
```
Comment (matteobettini) [inline torchrl/_utils.py:550]: Why are we still keeping the name unravel_keys? Cant we do all this and also stop using that plural in all the 2 projects?
I thought after this whole thing we were going to have:

- unravel_key
- unravel_key_list
- _unravel_tuple_key

with unravel_keys dead

**Review by matteobettini — APPROVED** [review]
LGTM, just one comment

File: torchrl/_utils.py (hunk 1 above)
Comment (vmoens) [reply torchrl/_utils.py:550]: this is just for this PR, it'll be used for 1 hour max
I just don't want the tests to break at any time bc that will be a pain to handle internally on fbcode
I can rename it unravel_key but that's beyond the point IMO
We could name it `_tmp_fun_that_we_will_remove_` fwiw

Comment (matteobettini) [reply torchrl/_utils.py:550]: ok got it go for it

Comment (vmoens) [reply torchrl/_utils.py:550]: btw we won't kill anything for bc reasons, we'll just keep both names (perhaps wrapping the plural in a deprecation decorator)

**Pull request merged**